src/args.rs

```diff
@@ -0,0 +1,219 @@
+        match s.to_lowercase().as_str() {
+            "grpc" => OtlpProtocol::Grpc,
+            "http" => OtlpProtocol::Http,
+            _ => {
+                error!(
+                    "Invalid OTLP protocol {}, defaulting to {}",
+                    s,
+                    OtlpProtocol::default()
+                );
+                OtlpProtocol::default()
+            }
+        }
+    }
+}
+
+impl OtlpProtocol {
+    pub fn default_endpoint(&self) -> &str {
+        match self {
+            OtlpProtocol::Grpc => "http://localhost:4317",
+            OtlpProtocol::Http => "http://localhost:4318",
+        }
+    }
+}
+
+#[derive(Debug, Clone, Copy, Default, PartialEq)]
+pub enum StdoutLogConfig {
+    #[default]
+    Full,
+    Compact,
+    Pretty,
+    JSON,
+    None,
+}
+
+impl Display for StdoutLogConfig {
+    fn fmt(&self, f: &mut std::fmt::Formatter<'_>) -> std::fmt::Result {
+        match self {
+            StdoutLogConfig::Full => write!(f, "full"),
+            StdoutLogConfig::Compact => write!(f, "compact"),
+            StdoutLogConfig::Pretty => write!(f, "pretty"),
+            StdoutLogConfig::JSON => write!(f, "json"),
+            StdoutLogConfig::None => write!(f, "none"),
+        }
+    }
+}
+
+impl From<String> for StdoutLogConfig {
+    fn from(s: String) -> Self {
+        match s.to_lowercase().as_str() {
+            "full" => StdoutLogConfig::Full,
+            "compact" => StdoutLogConfig::Compact,
+            "pretty" => StdoutLogConfig::Pretty,
+            "json" => StdoutLogConfig::JSON,
+            "none" => StdoutLogConfig::None,
+            _ => {
+                warn!(
+                    "Invalid trace format {}, defaulting to {}",
+                    s,
+                    StdoutLogConfig::default()
+                );
+                StdoutLogConfig::default()
+            }
+        }
+    }
+}
+
+#[derive(Debug, Clone)]
+pub struct OtlpExportConfig {
+    pub service_name: String,
+    pub traces: Option<(OtlpProtocol, String)>,
+    pub metrics: Option<(OtlpProtocol, String)>,
+    pub stdout_log: StdoutLogConfig,
+}
+
+impl From<Args> for OtlpExportConfig {
+    fn from(args: Args) -> Self {
+        let service_name = args.otlp_service_name;
+        let otlp_protocol = args.otlp_protocol;
+        let otlp_endpoint = args
+            .otlp_endpoint
+            .unwrap_or(otlp_protocol.default_endpoint().to_string());
+        let otlp_traces_endpoint = args.otlp_traces_endpoint.unwrap_or(otlp_endpoint.clone());
+        let otlp_metrics_endpoint = args.otlp_metrics_endpoint.unwrap_or(otlp_endpoint.clone());
+        let otlp_traces_protocol = args.otlp_traces_protocol.unwrap_or(otlp_protocol);
+        let otlp_metrics_protocol = args.otlp_metrics_protocol.unwrap_or(otlp_protocol);
+        let stdout_log = args.stdout_log;
+
+        OtlpExportConfig {
+            service_name,
+            traces: match args.otlp_export.contains(&OtlpExport::Traces) {
```

Comment:
otlp export is a list, it can contain traces or metrics or both. `Traces, metrics, logs` are the standard three data types that are exportable by opentelemetry, although as mentioned in several places, since we are using the recommended `tracing` crate for logging instead of something legacy, we dont need to handle logs separately